### PR TITLE
[FAD-48] Build Button component in library

### PR DESCRIPTION
**Pull request by v1n33th**
### Description

Modified button component to Figma designs and modified its usages

Figma:https://www.figma.com/file/g7Edk2JCOMLpRAw9v8NJ4c/FA-Product-Wireframe---working-file?node-id=29%3A1574

- [ ] `CHANGELOG` has been updated (_if appropriate_)
- [ ] Environment variables updated (_if appropriate_)

### CODE REVIEW

**Comment by v1n33th** [comment]
Working Image
chrome-capture-2022-8-7

File: packages/client3/components/design-system/button/button.tsx [hunk 1]
```diff
@@ -1,163 +1,93 @@
...
-          size === "sm"
-            ? "!p-0.5"
-            : size === "md"
-            ? "!p-1.5"
-            : size === "lg"
-            ? "!p-2.5"
-            : size === "xl"
-            ? "!p-[1.125rem]"
-            : null,
+          "rounded-[100px] ",
```
Comment (sachasmart-weavik) [inline packages/client3/components/design-system/button/button.tsx:53]: Would `rounded-full` work? 

Ref: https://tailwindcss.com/docs/border-radius#pill-buttons

File: packages/client3/tailwind.config.js [hunk 2]
```diff
@@ -37,6 +37,9 @@ module.exports = {
           3: "#F4F4F4",
           4: "#FDFDFD",
         },
+        accent1: {
+          100: "#BFED47",
+        },
```
Comment (sachasmart-weavik) [inline packages/client3/tailwind.config.js:42]: I think @luansievers  may be merging his updated branch shortly, so potentially rebase off of his when that is done?

**Review by sachasmart-weavik — APPROVED** [review]
Looks good man! 

I see Luan just merged his stuff, I'm going to sneak in another accent into your `tailwind.config.js`

File: packages/client3/tailwind.config.js (hunk 2 above)
Comment (sachasmart-weavik) [reply packages/client3/tailwind.config.js:42]: Doesn't look like this is in his stuff though, soo... nevermind?

File: packages/client3/components/design-system/button/button.tsx (hunk 1 above)
Comment (v1n33th) [reply packages/client3/components/design-system/button/button.tsx:53]: Yeah, I initially used `rounded-full` which resolves to `border-radius: 9999px`then changed it as in designs the required border radius was 100px. Saying that visually I couldn't find any differences in both the CSS styles.  Which do you think is better?

File: packages/client3/components/design-system/button/button.stories.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { ComponentStory, ComponentMeta } from "@storybook/react";
 
-import { Button, IconButton } from ".";
+import { Button } from ".";
 
 export default {
   title: "Components/Button",
```
Comment (luansievers) [inline packages/client3/components/design-system/button/button.stories.tsx:6]:
```suggestion
  title: "FAD/Components/Button",
```

File: packages/client3/components/design-system/button/button.tsx [hunk 4]
```diff
@@ -1,163 +1,94 @@
...
-            : size === "lg"
-            ? "!p-2.5"
-            : size === "xl"
-            ? "!p-[1.125rem]"
-            : null,
+          "rounded-[100px]",
+          "flex items-center",
+          children && "px-6",
+          {
+            "bg-accent-1 text-green-90 hover:bg-accent-1-shade disabled:opacity-40 disabled:hover:bg-accent-1":
```
Comment (luansievers) [inline packages/client3/components/design-system/button/button.tsx:57]: I think the opacity here is 30%

File: packages/client3/components/design-system/button/button.tsx [hunk 5]
```diff
@@ -1,163 +1,94 @@
...
+              buttonType == "primary",
+            "border border-accent-1 bg-white text-accent-1 hover:bg-accent-1/10 disabled:border-dark-80 disabled:text-dark-80 disabled:hover:bg-white":
+              buttonType == "secondary",
+            "bg-transparent text-white hover:bg-dark-80 disabled:text-dark-80 disabled:hover:bg-transparent":
+              buttonType == "tertiary",
```
Comment (luansievers) [inline packages/client3/components/design-system/button/button.tsx:62]: I think you gonna need to fix those classes with the dark theme, when I try secondary type, the hover class is wrong.

File: packages/client3/components/design-system/button/link-button.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  function LinkButton({ href, ...rest }, ref) {
+    return (
+      <NextLink passHref href={href}>
+        <a>
+          <Button {...rest} ref={ref} />
+        </a>
+      </NextLink>
+    );
+  }
+);
```
Comment (luansievers) [inline packages/client3/components/design-system/button/link-button.tsx:19]: I didn't quite understand the reason to this component, we already have a "Link" component..

**Review by luansievers — CHANGES_REQUESTED** [review]

File: packages/client3/components/design-system/button/button.stories.tsx (hunk 3 above)
Comment (v1n33th) [reply packages/client3/components/design-system/button/button.stories.tsx]: Done

File: packages/client3/components/design-system/button/button.tsx (hunk 4 above)
Comment (v1n33th) [reply packages/client3/components/design-system/button/button.tsx]: Sorry...My bad changed it

File: packages/client3/components/design-system/button/button.tsx (hunk 5 above)
Comment (v1n33th) [reply packages/client3/components/design-system/button/button.tsx]: Sorry..I think it's fixed now.

File: packages/client3/components/design-system/button/link-button.tsx (hunk 6 above)
Comment (v1n33th) [reply packages/client3/components/design-system/button/link-button.tsx:19]: Initially, the goldfinch button had a link button within it to handle links. So I just extracted it out as a separate component to handle it easily. The purpose of this component is to handle use cases where we need a page redirection from a button and also the current link component doesn't support having children.

Thoughts?

**Review by luansievers — APPROVED** [review]

File: packages/client3/components/design-system/button/link-button.tsx (hunk 6 above)
Comment (luansievers) [reply packages/client3/components/design-system/button/link-button.tsx:19]: thats fine I guess

**Pull request merged**